crates/server/src/admin/utils/auth.rs

```diff
@@ -283,6 +284,93 @@ pub fn validate_root_key_exists(
     drop(match root_key_result {
         Some(root_key) => root_key,
         None => {
+            if let WalletType::NEAR { network_id } = &req.wallet_metadata.wallet_type {
+                let wallet_address = match req.wallet_metadata.wallet_address.as_deref() {
+                    Some(address) => address,
+                    None => {
+                        return Err(ApiError {
```

Comment:
For all the places here that `return`, it would be more idiomatic to simply return them implicitly, and then use e.g. `?` at the end of the match assignment.